astropy/wcs/tests/test_profiling.py

```diff
@@ -7,14 +7,37 @@
 
 import numpy as np
 
+from ...tests.helper import pytest
 from ...utils.data import get_pkg_data_filenames, get_pkg_data_contents
 from ...utils.misc import NumpyRNGContext
 
 from ... import wcs
 
+#hdr_map_file_list = list(get_pkg_data_filenames("maps", pattern="*.hdr"))
 
-def test_maps():
-    def test_map(filename):
+# use the base name of the file, because everything we yield
+# will show up in the test name in the pandokia report
+hdr_map_file_list = [os.path.basename(fname) for fname in get_pkg_data_filenames("maps", pattern="*.hdr")]
+
+# Checking the number of files before reading them in.
+# OLD COMMENTS:
+# AFTER we tested with every file that we found, check to see that we
+# actually have the list we expect.  If N=0, we will not have performed
+# any tests at all.  If N < n_data_files, we are missing some files,
+# so we will have skipped some tests.  Without this check, both cases
+# happen silently!
+
+def test_read_map_files():
+    # how many map files we expect to see
+    n_map_files = 28
+
+    if len(hdr_map_file_list) != n_map_files:
+       assert False, (
```

Comment:
The construct was already in the file, but is truly odd, so might as well change:
```
assert len(hdr_map_file_list) == n_map_files, (<text as is>)
```
(and again below)